pkg/kubelet/cm/topologymanager/scope.go

```diff
@@ -148,11 +151,21 @@ func (s *scope) admitPolicyNone(pod *v1.Pod) lifecycle.PodAdmitResult {
 // It would be better to implement this function in topologymanager instead of scope
 // but topologymanager do not track providers anymore
 func (s *scope) allocateAlignedResources(pod *v1.Pod, container *v1.Container) error {
+	isPodLevelResourcesEnabled := utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources)
+
+	var errs []error
 	for _, provider := range s.hintProviders {
 		err := provider.Allocate(pod, container)
-		if err != nil {
+		if err != nil && isPodLevelResourcesEnabled {
+			errs = append(errs, err)
```

Comment:
What are the implications of not short circuiting here? Coupling this to the event emission seems risky.
I think @ffromani brought this up on the other PR too.